### PR TITLE
feat(alerts): Add tests for alerts

### PR DESCRIPTION
**Pull request by udhadpa**
Hello All, 
Please review my test for Alerts. 
It includes test cases as per the scenario sheet shared earlier.
Do let me know your feedback.
@mhomolak  @JasonLantz  @wyseguyonline  @umahaea 
Thanks,

### CODE REVIEW

File: src/main/java/origamiV2/origamiV2PageObjects/AlertsPageObjects.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        this.driver = driver;
+    }
+
+    public AlertsPageObjects(AppiumDriver appium) {
+        this.appium = appium;
+    }
+
+    public By successAlertBtn = By.id("success-alert");
+    public By errorAlertBtn = By.id("error-alert");
+    public By clearAlertBtn = By.id("clear-alert");
```
Comment (JasonLantz) [inline src/main/java/origamiV2/origamiV2PageObjects/AlertsPageObjects.java]: the "clear" functionality was put in by dev for demo purposes and not enumerated in the design spec....

File: src/test/java/origamiV2Tests/AlertsTest.java [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+        marginTop = commonUtils.getCSSValue(alert, "margin-top");
+        borderTopWidth = commonUtils.getCSSValue(alert, "border-top-width");
+        borderTopStyle = commonUtils.getCSSValue(alert, "border-top-style");
+        borderTopColor = commonUtils.getCSSValue(alert, "border-top-color");
+        bgColor = commonUtils.getCSSValue(alert, "background-color");
+
+        for (String cssProperty : paddings) {
+            padding = commonUtils.getCSSValue(alert, cssProperty);
+            isPadding = commonUtils.assertCSSProperties(cssProperty, padding, new String[]{"10px", "15px"});
+            if (isPadding == false) {
```
Comment (mhomolak) [inline src/test/java/origamiV2Tests/AlertsTest.java]: Could we change these conditionals to if(!isPadding) ?

File: test_suites/elements_sdk.xml [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 <suite name="Elements sdk V1 Test Suite" parallel="classes">
 
-    <parameter name="runEnv" value="sauce"></parameter>
+    <parameter name="runEnv" value="local"></parameter>
```
Comment (umahaea) [inline test_suites/elements_sdk.xml]: can we set this to "sauce"?

File: src/test/java/origamiV2Tests/AlertsTest.java [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+    }
+
+    @Test(testName = "Mobile : Click on 'X' icon", dataProvider = "Click on 'X' icon Test Data", groups = "mobile-regression")
+    private void clickOnXIconMobileTest(String alertType, By button, By alert, By icon) throws InterruptedException {
+        commonUtils.clickUsingJS(button, "mobile");
+        Thread.sleep(500);
+        commonUtils.clickUsingJS(icon, "mobile");
+        Thread.sleep(500);
+        isElemPresent = commonUtils.isElementsVisibleOnPage(alert, "mobile");
+        if (isElemPresent == true) {
```
Comment (mhomolak) [inline src/test/java/origamiV2Tests/AlertsTest.java]: Same thing here.. if (isElemPresent) does the same thing with less verbiage.

File: src/test/java/origamiV2Tests/AlertsTest.java [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+    @BeforeClass(alwaysRun = true)
+    private void beforeClass(String sauceBrowser, String mobile) {
+        browser = sauceBrowser;
+        this.mobile = mobile;
+    }
+
+    @DataProvider(name = "Success and Error Alerts Test Data")
+    public Object[][] getSuccessErrorAlertButtonTestData() {
+        return new Object[][]{
+                {alertsPgObj.successAlertBtn, "success-alert", By.id("alert-success-0"), "50px", "6px", "solid", new String[]{commonUtils.hex2Rgb("#038238"), commonUtils.hex2RgbWithoutTransparency("#038238")}, new String[]{commonUtils.hex2Rgb("#ffffff"), commonUtils.hex2RgbWithoutTransparency("#ffffff")}, By.xpath("//*[@id='alert-success-0']/strong"), By.xpath("//*[@id='alert-success-0']/span"), "8px"},
```
Comment (umahaea) [inline src/test/java/origamiV2Tests/AlertsTest.java]: why do we have to write the raw locators here , xpaths and ids? Can we put them into page objects class and use the variable instead? will help in the future for any changes to the locators.

File: src/test/java/origamiV2Tests/AlertsTest.java [hunk 6]
```diff
@@ -0,0 +1,313 @@
...
+            log.info("Font size of Text for " + alertType + " is not as per spec, actual " + textFontSize);
+        }
+        isTextLineHt = commonUtils.assertCSSProperties("line-height", textLineHt, new String[]{"22px", "22.000019073486328px"});
+        if (isTextLineHt == false) {
+            log.info("Line height of text for " + alertType + " is not as per spec, actual " + textLineHt);
+        }
+        isPaddingTop = commonUtils.assertValue(paddingTop, expPaddingTop, "Padding-bottom of text for " + alertType + " is not as per spec");
+
+        Assert.assertTrue(isMarginTop && isBorderTopWidth && isBorderTopStyle && isBorderTopColor && isBgColor && isTitleFontColor && isTitleFontSize && isTitleLineHt && isTextFontColor && isTextFontSize && isTextLineHt && isPaddingTop);
+        commonUtils.click(alertsPgObj.clearAlertBtn);
```
Comment (umahaea) [inline src/test/java/origamiV2Tests/AlertsTest.java]: why do we click after the assert has happened?

File: src/main/java/origamiV2/origamiV2PageObjects/AlertsPageObjects.java (hunk 1 above)
Comment (udhadpa) [inline src/main/java/origamiV2/origamiV2PageObjects/AlertsPageObjects.java]: Hi @JasonLantz  : I have completely removed the tests written for Clear buttons. Thank You for pointing it out.

File: src/test/java/origamiV2Tests/AlertsTest.java (hunk 2 above)
Comment (udhadpa) [inline src/test/java/origamiV2Tests/AlertsTest.java]: Thanks Mike for the suggestion. I have fixed these statements.

**Comment by umahaea** [comment]
Rest looks good to me. Great work!!

**Comment by udhadpa** [comment]
@umahaea  : Fixed tests as per your suggestion. Thanks

**Comment by umahaea** [comment]
Got max reviews. going to merge.

**Pull request merged**